pages/index.js

```diff
@@ -1,13 +1,54 @@
 import styled from 'styled-components';
 
-export default function Home() {
+import fsPromises from 'fs/promises';
+import path from 'path';
+import Head from 'next/head';
+import RecordFile from '../components/RecordFile';
+
+export async function getStaticProps() {
+  const filePath = path.join(process.cwd(), 'data-dummy.json');
+  const jsonData = await fsPromises.readFile(filePath);
```

Comment:
Also can't say too much about this part of the code to get the data since I don't know the syntax.

Just an idea what you could write alternatively:
- create a js file "recordService" for your record data. 
- create a variable and assing the data array to it.
- create a function in the recordService.js that returns the variable
- export the function so you can use it anywhere else
- import the function here under getStaticProps, save them in a variable and return them as props for you page.